src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -682,9 +880,9 @@ WorkerCreateShard(Oid relationId, int shardIndex, uint64 shardId, List *ddlComma
 		appendStringInfo(applyForeignConstraintCommand,
 						 WORKER_APPLY_INTER_SHARD_DDL_COMMAND, shardId, escapedSchemaName,
 						 referencedShardId, escapedReferencedSchemaName, escapedCommand);
+		appendStringInfoChar(applyForeignConstraintCommand, ';');
```

Comment:
Why are we adding ';' here and other places? Aren't we using join by delimiters in `CreateShardsOnWorkersViaExecutor()`?

Reply:
Removed it on the other branch, sorry a tricky review for you :)